### PR TITLE
Base+WindowServer: Support hover icons for title buttons

### PR DESCRIPTION
**Pull request by MacDue**
This PR adds support for `-hover.png` variants of the title button icons, which now allows themes that use `TitleButtonsIconOnly`
to show the hover state (though this feature is not strictly for use with `TitleButtonsIconOnly`). 

I've added some hover icons to the Coffee theme, though there's other themes (e.g. Cupertino & Light) that could make use of this. 

https://user-images.githubusercontent.com/11597044/166325084-4e016297-1c63-4160-ad71-0e6667951593.mp4

### CODE REVIEW

**Comment by djwisdom** [comment]
Nice! A breath of fresh air. Almost like SerenityOS **2.0**   :^)

File: Userland/Services/WindowServer/Button.cpp [hunk 1]
```diff
@@ -31,12 +31,18 @@ void Button::paint(Screen& screen, Gfx::Painter& painter)
...
         if (m_pressed)
             painter.translate(1, 1);
         painter.blit(icon_location, bitmap, bitmap.rect());
+    };
+
+    if (m_icon.hover_bitmap && m_hovered) {
+        paint_icon(m_icon.hover_bitmap);
+    } else if (m_icon.bitmap) {
+        paint_icon(m_icon.bitmap);
     }
```
Comment (linusg) [inline Userland/Services/WindowServer/Button.cpp:46]: No curly braces needed here

**Review by linusg — CHANGES_REQUESTED** [review]
Small style nitpick, please also run `optipng -strip all` over these images to save a couple more bytes :)

File: Userland/Services/WindowServer/Button.cpp (hunk 1 above)
Comment (MacDue) [reply Userland/Services/WindowServer/Button.cpp]: Braces fixed, and images optimised

Comment (MacDue) [reply Userland/Services/WindowServer/Button.cpp]: Btw could the linter enforce no braces for single-line if/else bodies?

Comment (linusg) [reply Userland/Services/WindowServer/Button.cpp]: AFAIK clang-format can't do that, unfortunately.

**Comment by nico** [comment]
Should this have added the same thing to LibGUI/AbstractThemePreview.cpp?

**Pull request merged**